### PR TITLE
Add Detailed Logging and Error Handling for Chart Download

### PR DESCRIPTION
**Pull request by godhanipayal**
Add Detailed Logging and Error Handling for Chart Download

Signed-off-by: Payal Godhani <godhanipayal@gmail.com>

**What this PR does / why we need it**: 

This PR adds detailed logging and error messages for chart download operations

- Errors are logged with context for easier troubleshooting.
- Defensive check for empty data.
- Early returns reduce nesting.
- Logging now uses slog.Error for actual errors, slog.Debug for process information

Enhances maintainability and debuggability of the chart download logic by providing clearer logs and handling edge cases gracefully.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/downloader/chart_downloader.go [hunk 1]
```diff
@@ -246,22 +246,31 @@ func (c *ChartDownloader) DownloadToCache(ref, version string) (string, *provena
...
+			slog.Error("failed to download chart from remote", "error", gerr, "url", u.String(), "ref", ref, "version", version)
 			return "", nil, gerr
 		}
 
 		// Generate the digest
 		if len(digest) == 0 {
-			digest32 = sha256.Sum256(data.Bytes())
+			// Defensive: check data.Bytes() is not nil/empty
+			bytes := data.Bytes()
+			if len(bytes) == 0 {
```
Comment (TerryHowe) [inline pkg/downloader/chart_downloader.go:264]: Does something else check for empty chart data or is it supposed to be valid?

The sh256 should work fine for ""

**Review by TerryHowe — COMMENTED** [review]
I'd like someone else to weigh in here that knows this code better, but I would think we'd want to slog.Debug these instead and let the higher levels do logging if any.

The other alternative would be wrap the errors instead to provide additional information.

File: pkg/downloader/chart_downloader.go (hunk 1 above)
Comment (godhanipayal) [reply pkg/downloader/chart_downloader.go:264]: Thanks @TerryHowe  for reviewing.  I checked the codebase and dint find any explicit checks for empty chart data right after downloading. There are other validations for chart contents like metadata and such.

You have a good point about sha256.Sum256 and it does work with empty data. I added this check as in this context, empty chart download likely means somehting went wrong like network issue or bad remote response. This check will make it fail fast and prevents unnecessary processing

**Comment by godhanipayal** [comment]
> I'd like someone else to weigh in here that knows this code better, but I would think we'd want to slog.Debug these instead and let the higher levels do logging if any.
> 
> The other alternative would be wrap the errors instead to provide additional information.

- Changed the log level to debug
